azurestack/resource_arm_loadbalancer_nat_rule.go

```diff
@@ -0,0 +1,302 @@
+		d.Set("enable_floating_ip", props.EnableFloatingIP)
+
+		if ipconfiguration := props.FrontendIPConfiguration; ipconfiguration != nil {
+			fipID, err := parseAzureResourceID(*ipconfiguration.ID)
+			if err != nil {
+				return err
+			}
+
+			d.Set("frontend_ip_configuration_name", fipID.Path["frontendIPConfigurations"])
+			d.Set("frontend_ip_configuration_id", ipconfiguration.ID)
+		}
+
+		if ipconfiguration := props.BackendIPConfiguration; ipconfiguration != nil {
+			d.Set("backend_ip_configuration_id", ipconfiguration.ID)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmLoadBalancerNatRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).loadBalancerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	loadBalancerID := d.Get("loadbalancer_id").(string)
+	armMutexKV.Lock(loadBalancerID)
+	defer armMutexKV.Unlock(loadBalancerID)
+
+	loadBalancer, exists, err := retrieveLoadBalancerById(loadBalancerID, meta)
+	if err != nil {
+		return fmt.Errorf("Error Getting LoadBalancer By ID: %+v", err)
+	}
+	if !exists {
+		d.SetId("")
+		return nil
+	}
+
+	_, index, exists := findLoadBalancerNatRuleByName(loadBalancer, d.Get("name").(string))
+	if !exists {
+		return nil
+	}
+
+	oldNatRules := *loadBalancer.LoadBalancerPropertiesFormat.InboundNatRules
+	newNatRules := append(oldNatRules[:index], oldNatRules[index+1:]...)
+	loadBalancer.LoadBalancerPropertiesFormat.InboundNatRules = &newNatRules
+
+	resGroup, loadBalancerName, err := resourceGroupAndLBNameFromId(d.Get("loadbalancer_id").(string))
+	if err != nil {
+		return fmt.Errorf("Error Getting LoadBalancer Name and Group: %+v", err)
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, loadBalancerName, *loadBalancer)
+	if err != nil {
+		return fmt.Errorf("Error Creating/Updating LoadBalancer %q (Resource Group %q) %+v", loadBalancerName, resGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for the completion of LoadBalancer updates for %q (Resource Group %q) %+v", loadBalancerName, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, loadBalancerName, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving LoadBalancer %q (Resource Group %q): %+v", loadBalancerName, resGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read LoadBalancer %q (resource group %q) ID", loadBalancerName, resGroup)
+	}
+
+	return nil
+}
+
+func expandAzureRmLoadBalancerNatRule(d *schema.ResourceData, lb *network.LoadBalancer) (*network.InboundNatRule, error) {
+
+	properties := network.InboundNatRulePropertiesFormat{
+		Protocol:     network.TransportProtocol(d.Get("protocol").(string)),
+		FrontendPort: utils.Int32(int32(d.Get("frontend_port").(int))),
+		BackendPort:  utils.Int32(int32(d.Get("backend_port").(int))),
+	}
+
+	if v, ok := d.GetOk("enable_floating_ip"); ok {
+		enableFloatingIP := v.(bool)
+		properties.EnableFloatingIP = utils.Bool(enableFloatingIP)
```

Comment:
These can be combined into: `properties.EnableFloatingIP = utils.Bool(v.(bool))`
 `